src/Npgsql.NodaTime/Internal/LegacyConverters.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using NodaTime;
+using Npgsql.Internal;
+using static Npgsql.NodaTime.Internal.NodaTimeUtils;
+
+namespace Npgsql.NodaTime.Internal;
+
+public class LegacyTimestampTzZonedDateTimeConverter : PgBufferedConverter<ZonedDateTime>
+{
+    readonly bool _dateTimeInfinityConversions;
+
+    public LegacyTimestampTzZonedDateTimeConverter(bool dateTimeInfinityConversions)
+        => _dateTimeInfinityConversions = dateTimeInfinityConversions;
+
+    public override bool CanConvert(DataFormat format, out BufferingRequirement bufferingRequirement)
+    {
+        bufferingRequirement = BufferingRequirement.FixedSize;
+        return base.CanConvert(format, out bufferingRequirement);
+    }
+
+    public override Size GetSize(SizeContext context, ZonedDateTime value, ref object? writeState)
+        => sizeof(long);
+
+    protected override ZonedDateTime ReadCore(PgReader reader)
+    {
+        throw new NotImplementedException("Need access to TimeZone connection parameter");
```

Comment:
As I think needing this state stinks hugely we should probably add an interface with just this timezone property to Npgsql.Internal which PgReader implements explicitly until we kill legacy support?

Reply:
There's maybe an easier variant here: we can assume TimeZone doesn't change throughout the connection lifetime, at which point maybe it's easy to just shove it into PgSerializerOptions (like TextEncoding)? What do you think?

Reply:
Good one, let's go with that. Should I do it?

Reply:
Sure, "it's infra" :)

Reply:
I've just added it, take a look whether the approach makes sense. I've made its parsing lazy, I don't like preventing connections from functioning when they might never need to use the timezone .

Reply:
![image](https://github.com/NinoFloris/npgsql/assets/1862641/60f015ee-c38c-4382-b139-dbf717e56292)